### DISCUSSION TITLE
Branch 7.1 in code tab dropdown menu

### DISCUSSION THREAD

**Discussion by stefanrueger** [avrdudes/avrdude#1283] [General]
Do we want only branch 7.1 in the branch dropdown of the code tab or both 7.0 and 7.1?

![branch7 1](https://user-images.githubusercontent.com/14282046/212377677-5c4f59e5-8f21-4592-96a3-313c8a259c80.png)

**Comment by MCUdude** [comment]
I'd prefer `7.1-branch` and `7.0-branch` to be visible

**Comment by mariusgreuel** [comment]
I'd remove the old branches altogether and stick with tags, unless there is actually work being done on a branch. Also, I suggest to not append '-branch' to a branch.

**Comment by mcuee** [comment]
Converted this into a conversation.

Since we are not working on the 7.0/7.1 codes any more, I do not see the need for 7.0 or 7.1 branch at all. I think we can just remove the 7.0-branch as well to avoid confusions.

I think our release model is to tag the git main for a release when it is ready. If we find major issues, then we just fix it and then tag a new release.

**Comment by stefanrueger** [comment]
I have used the 7.0-branch in the past to check it out as a reference. But yes, I am sure I can get to a specific release with the right git-fu, so no problem deleting the branches that signify versions. One more question and two bonus questions
 - Do we need the onlinedocs branch?
 - Is there an *easy* way to force the SHA of a release to be mnemonic, eg, 0701000...?
 - How can discussions be closed so they drop off the list (some discussions, such as this one will be spent at some point)

**Comment by mariusgreuel** [comment]
> Do we need the onlinedocs branch?

Jörg put the GitHub pages there. See https://github.com/avrdudes/avrdude/discussions/905#discussioncomment-2640899

> Is there an easy way to force the SHA of a release to be mnemonic

I think you want [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging).

Most commands in Git can work with anything that points to a commit, such as tag, branch, or the commit ID itself. For instance, you can do a `git checkout v7.1` to checkout that release (and a `git checkout main` to get back).

**Reply by mcuee** [reply]
> Most commands in Git can work with anything that points to a commit, such as tag, branch, or the commit ID itself. For instance, you can do a git checkout v7.1 to checkout that release (and a git checkout main to get back).

That is exactly the reason why we do not need a v7.0 or v7.1 branch.

The only reason if we need the v7.0 or v7.1 branch is that we would continue to carry out bug fix for the old version. But we do not.

**Reply by dl8dtl** [reply]
> The only reason if we need the v7.0 or v7.1 branch is that we would continue to carry out bug fix for the old version. But we do not.

Exactly. We have been doing a dot-dot release in the past once or twice, to immediately followup with a quick bugfix right after a release. Normally, I think concentrating development effort on the next major release makes more sense.
